### DISCUSSION TITLE
6 meter cat 6 for i2c

### DISCUSSION THREAD

**Discussion by sphen13** [tronikos/esphome-magnetometer-water-gas-meter#15] [Q&A] [answered]
so.. am i an idiot thinking a 6 meter cat 6 run will work? running off esp8266. i powered with 3.3v first and it seemed to run for a bit - then i stopped getting data. Now i switched the sensor to 5v and im getting data again. you think i need to engineer this differently? 

thanks for the help! this project is great!

**Comment by sphen13** [comment]
yeah so - that didnt work long - i may just need to power the esp with a POE splitter and get it closer :/

**Comment by tronikos** [comment, marked as answer]
I'm using 10m CAT6 without issues. Yes powering the sensor with 5V is more reliable. 

Also make sure you followed the README: "The ethernet cable has 4 twisted pairs of wires. Use any solid wire color for the 4 above pins. Tie the 4 white wires together with the GND solid wire."

**Comment by sphen13** [comment]
alright well - im confused now -- shortened the wire - same thing after x amount of minutes to an hour or two -- now i have a scope hooked up and i can see the i2c - seems good to me, but when i check it later there is no signal..... hmmm. so now i am going to swap out the wemos d1??

**Comment by sphen13** [comment]
thanks @tronikos for letting me know it works for you - i ended up testing abunch of things and i just rewired the cat6 cable - i must have done something wrong even though all looked good before, but now it seems to be stable. its been running for about 24 hours with no i2c lock up. now to make sure the calibration etc stays steady :) thanks!

**Comment by aneisch** [comment]
Chiming in to report that I'm using ~20m CAT6 (https://www.amazon.com/dp/B08P2RLBJH) with no issue.

**Reply by tronikos** [reply]
Good to hear! I updated the README.

**Comment by daelsc** [comment]
Is there a particular advantage to running a long CAT run vs. running power and putting the ESP out there and connecting back with WiFi?  I guess potentially better WiFI reception and less electronics to waterproof?

**Reply by tronikos** [reply]
Yes if you put the ESP in the water meter box you have to waterproof more electronics and WiFi will be bad due to the concrete and stainless steel lid typically used on water meter boxes. That was my backup plan if i2c over long wires wouldn't work for me.

**Comment by NateEaton** [comment]
I've been running over ~22 meter [CAT6](https://a.co/d/7yadRyy) successfully for about 3 days with the cable running along the edge of my yard. The ESP32 is running off a USB-C cable from an AC adapter, wired to the 5v and ground. The sensor is wired to the 3v and ground.

Yesterday, I buried the cable (it's shallow, right along the edge of the driveway) and afterwards when I restarted the ESP32, its log showed qmc5883l failed. I was careful not to crimp the cable while burying it so I was a bit stumped. 

This morning I re-read the README and realized when I wired the sensor I neglected to connect the extra white wires to the ground. I'm guessing that may be my problem? I don't have previous experience with outdoor cabling like this so not sure I understand the need for that step. 

The good news is that I over-engineered my setup with [waterproof RJ45 connectors](https://a.co/d/a9wujt2) so that I can disconnect the sensor from the cable in the water meter box and from the ESP32 housing up at the house, meaning I can rewire things without pulling the cable up. I'll go ahead and work on getting that done but wanted to check in here anyway.
